coalib/results/Diff.py

```diff
@@ -70,6 +72,91 @@ def from_string_arrays(cls, file_array_1, file_array_2, rename=False):
         return result
 
     @classmethod
+    def from_unified_diff(cls, unified_diff, original_file):
+        """
+        Creates a ``Diff`` object from given unified diff.
+
+        If the provided unified diff does not contain any patch,
+        the ``Diff`` object initialized from the original file is
+        returned.
+
+        :param unified_diff:  Unified diff string.
+        :param original_file: The contents of the original file
+                              (line-splitted).
+        :raises RuntimeError: Raised when the context lines or the
+                              lines to be removed do not match in
+                              the original file and the unified diff.
+        """
+        patch_set = PatchSet(unified_diff.splitlines())
+
+        diff = Diff(original_file)
+
+        if not patch_set:
+            return diff
+
+        file = patch_set[0]
+
+        for hunk in file:
+            file_line = hunk.source_start
+            hunk_iterator = iter(hunk)
+
+            try:
+                while True:
+                    line = next(hunk_iterator)
+                    source_code = str(line)[1:]
+                    if line.is_added:
+                        add_set = []
+                        # As ``Diff`` does not allow line additions to a
+                        # position more than one time, add all the
+                        # consecutive '+' lines at once.
+
+                        try:
+                            while line.is_added:
+                                source_code = str(line)[1:]
+                                add_set.append(source_code)
+                                line = next(hunk_iterator)
+
+                            diff.add_lines(file_line-1, add_set)
+                            source_code = str(line)[1:]
+
+                        except StopIteration:
+                            diff.add_lines(file_line-1, add_set)
+                            break
+
+                    if line.is_removed:
+                        if (source_code !=
+                                original_file[file_line-1].strip('\n')):
+                            raise RuntimeError(
+                                'The line to delete does not match with'
+                                'the line in the original file. \n'
+                                'Line to delete: {} \n'
+                                'Original line #{}: {}'.format(
```

Comment:
Up to you, but for exception messages I usually try to avoid `\n`, as this applies some kind of formatting to the messages and imo exception messages should be only information. I try to separate sentences with periods and commas instead of `\n`. But as we don't have some general guideline for this, up to you ;)